Include ext DB migration for containerized (#5113)
theforeman/foreman-documentation: guides/common/modules/proc_migrating-to-external-databases.adoc

= Migrating to external databases

[role="_abstract"]
Migrate internal PostgreSQL databases from {ProjectServer} to a dedicated external host when you need to reduce server load or manage the database independently.

ifdef::containerized[]
[NOTE]
====
{insights-iop} requires internal database mode.
You cannot migrate to an external database while {insights-iop} is enabled on {ProjectServer}.
====
endif::containerized[]

.Prerequisites
* You have installed and configured a PostgreSQL server on an external server.
* You have confirmed that {ProjectServer} currently uses internal databases.

> For more information, see xref:determining-whether-your-{project-context}-server-uses-internal-or-external-databases[Determining whether your {ProjectServer} uses internal or external databases].

Reviewer: ```suggestion
For more information, see xref:determining-whether-your-{project-context}-server-uses-internal-or-external-databases[].
```

It's an xrefs, so we generally omit titles.

* You have scheduled a maintenance window.
ifdef::containerized[]
The `{foremanctl} backup` command stops all {Project} services during the backup process.

> endif::containerized[]

Reviewer: ```suggestion
endif::[]
```

.Procedure
ifdef::containerized[]
. On {ProjectServer}, create a directory for the backup:
+
[options="nowrap", subs="+quotes,attributes"]
----
# mkdir /var/_My_Migration_Backup_Directory_
----
. Back up the internal databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
ifdef::foreman-el,foreman-deb[]
# {foremanctl} backup /var/_My_Migration_Backup_Directory_
endif::[]
ifndef::foreman-el,foreman-deb[]
# {foremanctl} backup /var/_My_Migration_Backup_Directory_ \
--skip-pulp-content
endif::[]

> ----

Author: @LadislavVasina1 This argument for katello-enabled builds doesn't look right. This [readme](https://github.com/theforeman/foremanctl/blob/354c6c922f88117f1b2c7f02c854a76beb002f6b/docs/user/backup-restore.md#optional) says that `--skip-pulp-content` is for debugging purposes only. So perhaps the "production" command shouldn't use it and the backup command should be the same for both katello-enabled and katello-disabled builds?

Reviewer 2: @Lennonka Honestly I don't know why it became optional. 
In [6.19 docs](https://docs.redhat.com/en/documentation/red_hat_satellite/6.19/html/administering_red_hat_satellite/migrating_from_internal_databases_to_external_databases_admin#migrating-to-external-databases_admin) its does not mention it as optional, but I agree that if the foremanctl readme says it's optional, it should reflect the readme in documentation so that customers have that information.

Reviewer 3: Isn't foremanctl for katello-based builds only anyway? I thought that was the case originally at least; has that changed now?

Author: I don't think it changed, we probably just didn't fully realize that was the case. Thank you for mentioning that, Anet.

cc @maximiliankolb I'm going to undo the separation of containerized stuff for katello and non-katello builds, ok?

Reviewer: I saw your question in Matrix. Maybe give it some more time and start on Monday? 

But yeah, in general, if `foremanctl` is not intended for non-Katello deployments, then some of my comment were certainly wrong.

Author: Yes, I'll wait for a reply.

Reviewer 4: > Isn't foremanctl for katello-based builds only anyway? I thought that was the case originally at least; has that changed now?

That was never the intention, only a practical idea based on limited time. https://github.com/theforeman/foremanctl/pull/511 and https://github.com/theforeman/foremanctl/pull/527 are IMHO intended to be completed. Only then can we fully claim feature parity with package-based installations.

+
The backup stops all {Project} services, creates database dumps, and then restarts the services.

> The command creates a timestamped subdirectory, for example `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`, that contains `foreman.dump`, `candlepin.dump`, and `pulp.dump` files.

Reviewer: This line depends on the build flavor:

```suggestion
ifdef::foreman-el,foreman-deb[]
The command creates a timestamped subdirectory, for example, `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`, that contains the `foreman.dump` file.
endif::[]
ifdef::katello,orcharhino,satellite[]
The command creates a timestamped subdirectory, for example, `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`, that contains `foreman.dump`, `candlepin.dump`, and `pulp.dump` files.
endif::[]
```

I also added a comma after "for example". What do you think?

+
[NOTE]
====

> The backup fails if Foreman or Pulp tasks are running.

Reviewer: ```suggestion
The backup fails if {Project} tasks are running.
```

I suggest to simplify this unless there's a technical reason to mention those two services. If there is, then we should hide Pulp for non-Katello builds.

Wait for active tasks to finish before you run the backup, or add the `--wait-for-tasks` flag to wait for running tasks to complete.
====
. Verify the backup:
+
[options="nowrap", subs="+quotes,attributes"]
----
# ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-\*/
# cat /var/_My_Migration_Backup_Directory_/foreman-backup-*/metadata.yml
# ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-\*/*.dump
----
+

> Confirm that the timestamped backup directory contains `foreman.dump`, `candlepin.dump`, `pulp.dump`, and `metadata.yml` files.

Reviewer: Similar to the comment above: please hide candlepin and pulp for non-Katello builds.

= Migrating to external databases

[role="_abstract"]
Migrate internal PostgreSQL databases from {ProjectServer} to a dedicated external host when you need to reduce server load or manage the database independently.

[NOTE]
====
{insights-iop} requires internal database mode.
You cannot migrate to an external database while {insights-iop} is enabled on {ProjectServer}.
====

.Prerequisites
* You have installed and configured a PostgreSQL server on an external server.
* You have confirmed that {ProjectServer} currently uses internal databases.
For more information, see xref:determining-whether-your-{project-context}-server-uses-internal-or-external-databases[].
* You have scheduled a maintenance window.
ifdef::containerized[]
The `{foremanctl} backup` command stops all {Project} services during the backup process.
endif::[]

.Procedure
ifdef::containerized[]
. On {ProjectServer}, create a directory for the backup:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # mkdir /var/_My_Migration_Backup_Directory_

Reviewer: ```suggestion
# mkdir /var/_My_Migration_Backup_Directory_/
```

I suggest adding a trailing slash for the directory.

----
. Back up the internal databases:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # {foremanctl} backup /var/_My_Migration_Backup_Directory_

Reviewer: ```suggestion
# {foremanctl} backup /var/_My_Migration_Backup_Directory_/
```

Migrate internal PostgreSQL databases from {ProjectServer} to a dedicated external host when you need to reduce server load or manage the database independently.

ifdef::containerized[]
[NOTE]
====
{insights-iop} requires internal database mode.
You cannot migrate to an external database while {insights-iop} is enabled on {ProjectServer}.
====
endif::containerized[]

.Prerequisites
* You have installed and configured a PostgreSQL server on an external server.
* You have confirmed that {ProjectServer} currently uses internal databases.
For more information, see xref:determining-whether-your-{project-context}-server-uses-internal-or-external-databases[Determining whether your {ProjectServer} uses internal or external databases].
* You have scheduled a maintenance window.
ifdef::containerized[]
The `{foremanctl} backup` command stops all {Project} services during the backup process.
endif::containerized[]

.Procedure
ifdef::containerized[]
. On {ProjectServer}, create a directory for the backup:
+
[options="nowrap", subs="+quotes,attributes"]
----
# mkdir /var/_My_Migration_Backup_Directory_
----
. Back up the internal databases:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # {foremanctl} backup /var/_My_Migration_Backup_Directory_ \

Reviewer: This shows a trailing backslash for Foreman on Deb/EL.

ifndef::foreman-el,foreman-deb[]
--skip-pulp-content
endif::[]
----
+
The backup stops all {Project} services, creates database dumps, and then restarts the services.
The command creates a timestamped subdirectory, for example `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`, that contains `foreman.dump`, `candlepin.dump`, and `pulp.dump` files.
+
[NOTE]
====
The backup fails if Foreman or Pulp tasks are running.
Wait for active tasks to finish before you run the backup, or add the `--wait-for-tasks` flag to wait for running tasks to complete:

[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} backup /var/_My_Migration_Backup_Directory_ \
--wait-for-tasks \
ifndef::foreman-el,foreman-deb[]

> --skip-pulp-content

Reviewer: same as above. Here, you could flip the order to mitigate this.

= Migrating to external databases

[role="_abstract"]
Migrate internal PostgreSQL databases from {ProjectServer} to a dedicated external host when you need to reduce server load or manage the database independently.

> ifdef::containerized[]

Reviewer 4: I got curious if we ever supported IoP with external database. The old installer had `--iop-database-host` but we never documented that so I don't think we ever supported external DB with IoP. In other words:

```suggestion
```

= Migrating to external databases

[role="_abstract"]
Migrate internal PostgreSQL databases from {ProjectServer} to a dedicated external host when you need to reduce server load or manage the database independently.

.Prerequisites
* You have installed and configured a PostgreSQL server on an external server.
* You have confirmed that {ProjectServer} currently uses internal databases.
For more information, see xref:determining-whether-your-{project-context}-server-uses-internal-or-external-databases[Determining whether your {ProjectServer} uses internal or external databases].
* You have scheduled a maintenance window.
ifdef::containerized[]
The `{foremanctl} backup` command stops all {Project} services during the backup process.
endif::containerized[]

ifdef::containerized[]
[NOTE]
====

> The Insights Operating Platform (IOP) feature requires internal database mode.

Reviewer 2: ```suggestion
The Insights On Premise (IOP) feature requires internal database mode.
```

----
# mkdir /var/_My_Migration_Backup_Directory_
----
. Back up the internal databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} backup /var/_My_Migration_Backup_Directory_ \
ifndef::foreman-el,foreman-deb[]
--skip-pulp-content \
endif::[]
----
+
The backup stops all {Project} services, creates database dumps, and then restarts the services.
The command creates a timestamped subdirectory, for example `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`, that contains `foreman.dump`, `candlepin.dump`, and `pulp.dump` files.
+
[NOTE]
====
The backup fails if Foreman or Pulp tasks are running.
Wait for active tasks to finish before you run the backup, or add the `--wait-for-tasks` flag to wait for running tasks to complete:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} backup /var/_My_Migration_Backup_Directory_ --wait-for-tasks \
ifndef::foreman-el,foreman-deb[]
--skip-pulp-content \
endif::[]
----
====
. Verify the backup:
+
[options="nowrap", subs="+quotes,attributes"]
----
# ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-*/
# cat /var/_My_Migration_Backup_Directory_/foreman-backup-*/metadata.yml

> # ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-*/*.dump

Reviewer 2: ```suggestion
# ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-TIMESTAMP/
# cat /var/_My_Migration_Backup_Directory_/foreman-backup-TIMESTAMP/metadata.yml
# ls -lh /var/_My_Migration_Backup_Directory_/foreman-backup-TIMESTAMP/*.dump
```
I would add `TIMESTAMP` here so it matches the directory naming from above `/var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/`

Reviewer 2: Hmm, well yes `*` works ofcource too :D, but I am not seeing it in https://theforeman-foreman-documentation-preview-pr-5113.surge.sh/nightly/Administering_Project/index-satellite.html#Migrating_from_Internal_Databases_to_External_Databases_admin 

<img width="651" height="101" alt="Image" src="https://github.com/user-attachments/assets/82cf7fc9-e9ac-4f69-8f9f-38686d5c1ba0" />
@Lennonka is that some code snippet formatting issue you can fix so that the `*` is displayed?

Author: I think I will just have to escape it. I'll take a look at it later. Thank you!

endif::[]
/var/_My_Migration_Backup_Directory_
----
endif::[]
. Transfer the data to the new external databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
ifdef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/candlepin.dump

> PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/pulp.dump

Reviewer: Should we replace "foreman-backup" with "{project-context}-backup"? IMO yes unless there's a technical reason.

Reviewer: Could also be a follow-up issue.

Author: @adamruzicka Can you, please, confirm that the folder name will be branded, ie. `/var/backup/satellite-backup-TIMESTAMP/`?

Reviewer 5: I'm not aware of any plans, all I can speak to is the current state. As of now, [foremanctl uses a hardcoded non-branded variant](https://github.com/theforeman/foremanctl/blob/master/src/roles/backup/tasks/main.yaml#L8).

@sjha4 Hi, I saw your name on some of the backup/restore issues. Could you chime in here please?

Author: Keeping the hard-coded value in this PR.

endif::[]
endif::[]
ifndef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/pulpcore.dump
endif::[]
endif::[]
----
ifdef::containerized[]
+

> Replace `foreman-backup-_TIMESTAMP_` with the timestamp directory that the backup created.

Reviewer: same as above.

endif::[]
ifndef::foreman-el,foreman-deb[]
--foreman-database-password _Foreman_Password_ \
--candlepin-database-name candlepin \
--candlepin-database-user candlepin \
--candlepin-database-password _Candlepin_Password_ \
--pulp-database-name pulpcore \
--pulp-database-user pulp \
--pulp-database-password _Pulpcore_Password_
endif::[]
----
+
For a complete list of external database parameters, including optional SSL connection options, see {InstallingServerDocURL}deploy-{project-context}-server-configuration-with-an-external-database[Deploy {ProjectServer} configuration with an external database] in _{InstallingServerDocTitle}_.
. Verify that {ProjectServer} uses the external databases and that {Project} is operational:
+
[options="nowrap", subs="+quotes,attributes"]
----
# grep database_mode /var/lib/foremanctl/parameters.yaml
# {foremanctl} health
$ hammer ping
----
+
Confirm that `database_mode` is set to `external`, that `{foremanctl} health` reports no failures, and that `hammer ping` returns a successful response before you remove the internal PostgreSQL deployment.
. Remove the obsolete internal PostgreSQL deployment on {ProjectServer}:
+

> The internal PostgreSQL database runs as a Podman container managed by the `postgresql.service` systemd unit.

Reviewer: a bit similar to this: Is "Podman" required here?

Author: ```suggestion
The internal PostgreSQL database runs as a container managed by the `postgresql.service` systemd unit.
```

endif::[]
/var/_My_Migration_Backup_Directory_
----
endif::containerized[]
. Transfer the data to the new external databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
ifdef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/pulp.dump
endif::[]

> endif::containerized[]

Reviewer: ```suggestion
endif::[]
```

ifndef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/pulpcore.dump
endif::[]
endif::containerized[]
----
ifdef::containerized[]
+
Replace `foreman-backup-_TIMESTAMP_` with the timestamp directory that the backup created.
endif::containerized[]

> ifdef::containerized[]

Reviewer: ```suggestion
```

line 110 should suffice.

. Verify that {ProjectServer} uses the external databases and that {Project} is operational:
+
[options="nowrap", subs="+quotes,attributes"]
----
# grep database_mode /var/lib/foremanctl/parameters.yaml
# {foremanctl} health
$ hammer ping
----
+
Confirm that `database_mode` is set to `external`, that `{foremanctl} health` reports no failures, and that `hammer ping` returns a successful response before you remove the internal PostgreSQL deployment.
. Remove the obsolete internal PostgreSQL deployment on {ProjectServer}:
+
The internal PostgreSQL database runs as a Podman container managed by the `postgresql.service` systemd unit.
+
[options="nowrap", subs="+quotes,attributes"]
----
# systemctl stop postgresql.service
# systemctl disable postgresql.service
# rm -rf /var/lib/pgsql/data
----
+
[NOTE]
====
When you switch to external databases, `{foremanctl} deploy` does not remove Podman quadlet unit files under `/etc/containers/systemd/`, such as `postgresql.container` and `postgresql.image`, or Podman secrets, such as `postgresql-admin-password`.
Stopping the service and removing `/var/lib/pgsql/data` disables the internal database and frees the data volume.
====

> endif::containerized[]

Reviewer: ```suggestion
endif::[]
```

ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/foreman-backup-_TIMESTAMP_/pulp.dump
endif::[]
endif::[]
ifndef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/pulpcore.dump
endif::[]
endif::[]
----
ifdef::containerized[]
+

> Replace `foreman-backup-_TIMESTAMP_` with the timestamp directory that the backup created.

Reviewer: ```suggestion
Replace `foreman-backup-_TIMESTAMP_` with the timestamped directory that the backup created.
```

To align with lines 44 and 65.

endif::[]
endif::containerized[]
ifndef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/pulpcore.dump
endif::[]
endif::containerized[]
----
ifdef::containerized[]
+
Replace `foreman-backup-_TIMESTAMP_` with the timestamp directory that the backup created.
endif::containerized[]

ifdef::containerized[]
. Use the `{foremanctl} deploy` command to update {Project} to point to the new external databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--database-mode external \
--database-host _postgres.example.com_ \
--database-port 5432 \
--foreman-database-name foreman \
--foreman-database-user foreman \
--foreman-database-password _Foreman_Password_ \

> ifndef::foreman-el,foreman-deb[]

Reviewer: same issue as above.

endif::[]
endif::[]
ifndef::containerized[]
PGPASSWORD='_Foreman_Password_' pg_restore --host=_postgres.example.com_ --username=foreman --dbname=foreman < /var/_My_Migration_Backup_Directory_/foreman.dump
ifndef::foreman-el,foreman-deb[]
PGPASSWORD='_Candlepin_Password_' pg_restore --host=_postgres.example.com_ --username=candlepin --dbname=candlepin < /var/_My_Migration_Backup_Directory_/candlepin.dump
PGPASSWORD='_Pulpcore_Password_' pg_restore --host=_postgres.example.com_ --username=pulp --dbname=pulpcore < /var/_My_Migration_Backup_Directory_/pulpcore.dump
endif::[]
endif::[]
----
ifdef::containerized[]
+
Replace `foreman-backup-_TIMESTAMP_` with the timestamp directory that the backup created.
. Use the `{foremanctl} deploy` command to update {Project} to point to the new external databases:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--database-mode external \
--database-host _postgres.example.com_ \
--database-port 5432 \
--foreman-database-name foreman \
--foreman-database-user foreman \
ifdef::foreman-el,foreman-deb[]
--foreman-database-password _Foreman_Password_

> endif::[]

Reviewer 4: In the old migration tool we listed foreman last to avoid the ifdef

ifndef::foreman-el,foreman-deb[]
--foreman-database-password _Foreman_Password_ \
--candlepin-database-name candlepin \
--candlepin-database-user candlepin \
--candlepin-database-password _Candlepin_Password_ \
--pulp-database-name pulpcore \
--pulp-database-user pulp \
--pulp-database-password _Pulpcore_Password_
endif::[]
----
+
For a complete list of external database parameters, including optional SSL connection options, see {InstallingServerDocURL}deploy-{project-context}-server-configuration-with-an-external-database[Deploy {ProjectServer} configuration with an external database] in _{InstallingServerDocTitle}_.
. Verify that {ProjectServer} uses the external databases and that {Project} is operational:
+
[options="nowrap", subs="+quotes,attributes"]
----
# grep database_mode /var/lib/foremanctl/parameters.yaml
# {foremanctl} health

> $ hammer ping

Reviewer 4: Doesn't `health` already contain the same output as `hammer ping`?

----
+
Confirm that `database_mode` is set to `external`, that `{foremanctl} health` reports no failures, and that `hammer ping` returns a successful response before you remove the internal PostgreSQL deployment.
. Remove the obsolete internal PostgreSQL deployment on {ProjectServer}:
+
The internal PostgreSQL database runs as a container managed by the `postgresql.service` systemd unit.
+
[options="nowrap", subs="+quotes,attributes"]
----

> # systemctl stop postgresql.service
> # systemctl disable postgresql.service

Reviewer 4: ```suggestion
# systemctl disable --now postgresql.service
```

+
If you run {ProjectServer} on Debian 12, you have to adjust the paths of PostgreSQL config and data directories which depend on the PostgreSQL version.
endif::[]

> endif::containerized[]

Reviewer: ```suggestion
endif::[]
```
